exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/transactions/experimental/Suspended.kt

```diff
@@ -59,11 +67,23 @@ suspend fun <T> newSuspendedTransaction(
         suspendedTransactionAsyncInternal(true, statement).await()
     }
 
-suspend fun <T> Transaction.suspendedTransaction(context: CoroutineContext? = null, statement: suspend Transaction.() -> T): T =
+/**
+ * Calls the specified suspending [statement], suspends until it completes, and returns the result.
+ *
+ * The resulting `TransactionScope` is derived from the current `coroutineContext` if the latter already holds [this] `Transaction`;
+ * otherwise, a new scope is created using [this] `Transaction` and a given coroutine [context].
+ */
+suspend fun <T> Transaction.withSuspendTransaction(context: CoroutineContext? = null, statement: suspend Transaction.() -> T): T =
     withTransactionScope(context, this, db = null, transactionIsolation = null) {
         suspendedTransactionAsyncInternal(false, statement).await()
     }
```

Comment:
Does this need to come with some sort of intermediary that prevents breaking changes? Something like this:
```kt
@Deprecated(
    "Please rename to withSuspendTransaction",
    ReplaceWith("withSuspendTransaction(context, statement)"),
    DeprecationLevel.WARNING
)
suspend fun <T> Transaction.suspendedTransaction(context: CoroutineContext? = null, statement: suspend Transaction.() -> T): T =
    withSuspendTransaction(context, statement)
```

Same question for `transaction` that has had its parameters removed.